### ISSUE TITLE
iOS Postbuild process doesn't account for pre-existing entitlement files

### ISSUE THREAD

**Issue by NumaNumaNuma** [OneSignal/OneSignal-Unity-SDK#148]
**Description:**

If another entitlement file is already present in the xcode project, the build setting rule just appends the OneSignal one, causing a build error due to incorrect path, eg. "file1 file2"

It should merge the files if possible.

**Environment**

OneSignal 2.75 release
Unity 2018.2.8f
iOS12.0
XCode 10.0

**Steps to Reproduce Issue:**

1. Add entitlement file and build setting rule to your xcode project
2. Build Unity project with OneSignal package
3. build xcode project

**Anything else:**

**Comment by jkasten2** [comment]
@NumaNumaNuma Thanks for the details.

@Nightsd01 Could you look into this one?

**Comment by Nightsd01** [comment]
@NumaNumaNuma Thanks for pointing this out, we've resolved this in a recent PR (#150) and will be releasing an update shortly

**Comment by shutodev** [comment]
@Nightsd01 A similar problem is occurring in my environment.

**Environment**
- OneSignal 2.7.6 release + 00cd919 + f60a36a
- Unity 2018.2.6f1
- iOS 11.4
- Xcode 9.4.1

And PR #150 looks insufficient to solve this problem.
I actually tried to merge those commits into my branch, but the build error was still ongoing.

The cause of the build error is as follows: If the entitlements file already exists (for example, if `CODE_SIGN_ENTITLEMENTS` already has the value `file1.entitlements` ), the following code sets the value of `CODE_SIGN_ENTITLEMENTS` to like `file1.entitlements file2.entitlements` , It was that two values ​​were set.

https://github.com/OneSignal/OneSignal-Unity-SDK/blob/f60a36a940397dd33121712de40e7194d5a89199/OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs#L130

When an entitlements file already exists, adding a value to file1.entitlements in consideration of the contents of the existing file successfully built.

In conclusion, the real solution seems to be "if the entitlements file already exists, merge the contents of the existing entitlements file and save it as a file with the same name".

I'm a beginner in iOS, so those may contain something wrong.

**Comment by jkasten2** [comment]
@shutodev If there is an existing file it will be read so the existing elements won't be lost.
https://github.com/OneSignal/OneSignal-Unity-SDK/blob/f60a36a940397dd33121712de40e7194d5a89199/OneSignalExample/Assets/OneSignal/Editor/PostProcessBuildPlayer_iOS.cs#L107-L108

This merging only happens if your existing `.elements` file is your main target name which we use `PBXProject.GetUnityTargetName()` to get this name for clarification.

Did you manually add a new `.entitlements` to your project? Or did it come from another plugin? Or possibly did you rename your project?

@NumaNumaNuma Please try our or latest [OneSignal Unity SDK 2.7.8](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/2.7.8). If you still see an issue please also provide the details above about your pre-existing .elements` file.

**Comment by jkasten2** [comment]
@NumaNumaNuma Thanks for reporting this issue
@shutodev Thanks for the extra details and digging into the SDK code

This issue is fixed in PR #251 for Unity 2018.2 by reading `CODE_SIGN_ENTITLEMENTS` from the existing Xcode project and making only the necessary modifications.

This will be included in the next release
